source/MRMesh/MRCone3.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+
+// Base class for cone parameterization
+
+template <typename T>
+class Cone3
+{
+public:
+    Cone3()
+    {}
+
+    Cone3( const Line3<T>& inAxis, T inAngle, T inLength )
```

Comment:
I suggest adding a few comments:
1. The start of `inAxis` is the apex or the center of the base?
2. Is it ok if `inAxis` is not normalized? I'd either add a comment that it's allowed, or add `.normalized()` if not allowed.
3. Is `inAngle` a half-angle or a full angle?

Also at least to me `Line3` looks a bit confusing here, I'd pass and store two `Vector3` directly, but this is up to you.